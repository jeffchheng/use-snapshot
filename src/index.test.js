import React, { useState } from 'react';
import { render, fireEvent } from '@testing-library/react';

import {
  SnapshotProvider,
  useSnapshot,
  useStateSnapshot,
} from '.';

function State() {
  const state = useStateSnapshot();
  return <div>Stringified State: {JSON.stringify(state)}</div>;
}

function Foo() {
  useSnapshot("Foo", "Foo", true);
  return <div>Foo will snapshot</div>;
}

function Bar() {
  useSnapshot("Bar", "Bar", true);
  return <div>Bar will snapshot</div>;
}

function Baz() {
  useSnapshot("Baz", "Baz", false);
  return <div>Baz will not snapshot</div>;
}

function App() {
  const [showState, setShowState] = useState(false);
  return (
    <SnapshotProvider>
      <Foo />
      <Bar />
      <Baz />
      <button onClick={() => setShowState(state => !state)}>Snapshot</button>
      {showState && <State />}
    </SnapshotProvider>
  );
}

describe('use-snapshot', () => {
  it('can snapshot state with useSnapshot and provide it via useStateSnapshot', () => {
    const { container, getByText, rerender } = render(<App />);
    fireEvent.click(getByText('Snapshot'));
    expect(getByText(/Stringified State/i)).toMatchSnapshot();
  });
});
