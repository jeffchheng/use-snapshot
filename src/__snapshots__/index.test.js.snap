// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`use-snapshot can snapshot state with useSnapshot and provide it via useStateSnapshot 1`] = `
<div>
  Stringified State: 
  {"Foo":"Foo","Bar":"Bar","DynamicKey-1":"StaticValue","DynamicShouldSave":null}
</div>
`;

exports[`use-snapshot handles dynamic updates to keys by removing the previous key from state 1`] = `
<div>
  Stringified State: 
  {"Foo":"Foo","Bar":"Bar","DynamicKey-1":"StaticValue","DynamicShouldSave":null}
</div>
`;

exports[`use-snapshot handles dynamic updates to keys by removing the previous key from state 2`] = `
<div>
  Stringified State: 
  {"Foo":"Foo","Bar":"Bar","DynamicShouldSave":null,"DynamicKey-2":"StaticValue"}
</div>
`;

exports[`use-snapshot handles dynamic updates to keys by removing the previous key from state 3`] = `
<div>
  Stringified State: 
  {"Foo":"Foo","Bar":"Bar","DynamicShouldSave":null,"DynamicKey-3":"StaticValue"}
</div>
`;

exports[`use-snapshot handles dynamic updates to shouldSave by removing or adding to state 1`] = `
<div>
  Stringified State: 
  {"Foo":"Foo","Bar":"Bar","DynamicKey-1":"StaticValue","DynamicShouldSave":null}
</div>
`;

exports[`use-snapshot handles dynamic updates to shouldSave by removing or adding to state 2`] = `
<div>
  Stringified State: 
  {"Foo":"Foo","Bar":"Bar","DynamicKey-1":"StaticValue"}
</div>
`;

exports[`use-snapshot handles dynamic updates to shouldSave by removing or adding to state 3`] = `
<div>
  Stringified State: 
  {"Foo":"Foo","Bar":"Bar","DynamicKey-1":"StaticValue","DynamicShouldSave":null}
</div>
`;

exports[`use-snapshot handles dynamic updates to shouldSave by removing or adding to state 4`] = `
<div>
  Stringified State: 
  {"Foo":"Foo","Bar":"Bar","DynamicKey-1":"StaticValue"}
</div>
`;

exports[`use-snapshot updates state values when they change 1`] = `
<div>
  Stringified State: 
  {"Foo":"Foo","Bar":"Bar","DynamicKey-1":"StaticValue","DynamicShouldSave":null}
</div>
`;

exports[`use-snapshot updates state values when they change 2`] = `
<div>
  Stringified State: 
  {"Foo":"Foo","Bar":"Bar","DynamicKey-1":"StaticValue","DynamicShouldSave":null,"input":"whatevs"}
</div>
`;

exports[`use-snapshot updates state values when they change 3`] = `
<div>
  Stringified State: 
  {"Foo":"Foo","Bar":"Bar","DynamicKey-1":"StaticValue","DynamicShouldSave":null}
</div>
`;
